book/src/proptest/tips-and-best-practices.md

```diff
@@ -0,0 +1,41 @@
+# Tips and Best Practices
+
+## Performance
+
+### Setting `opt-level`
+Both the proptest crate and the random number generator it uses can be CPU intensive. If you are
+generating a lot of cases you may see a significant performance improvement by setting the `opt-level`
+to `3` in your `Cargo.toml` file:
+
+```toml
+[profile.test.package.proptest]
+opt-level = 3
+
+[profile.test.package.rand_chacha]
+opt-level = 3
+```
+
+### Reusing mutable resources 
+Sometimes you may want to reuse mutable resources across individual cases. For example, you may want
+to reuse a database connection or a file handle to avoid the overhead of opening and closing it for
+each case. Because the `proptest!` macro (when used with closure-style invocation) requires a `Fn`, you need to wrap your state in a `RefCell`:
+
+```rust,mdbook-runnable
+use std::cell::RefCell;
+
+# struct MyState {};
```

Comment:
`MyState` isn't actually used in the test it looks like. In terms of communicating what's happening, i'm not sure the `My` prefix is that helpful either and I'm guessing if `State` has `connect` it's really `ConnectionPool`? what do you think about using that name?

Reply:
Looking at this now, I think this example is broken. I'll revisit.